src/Atmos/Model/AtmosModel.jl

```diff
@@ -898,6 +906,264 @@ function numerical_flux_first_order!(
+    _e_int_v0 = e_int_v0(param_set)
+    Φ = gravitational_potential(balance_law, state_auxiliary⁻)
+
+    ρ⁻ = state_prognostic⁻.ρ
+    ρu⁻ = state_prognostic⁻.ρu
+    ρe⁻ = state_prognostic⁻.ρe
+    ρq_tot⁻ = state_prognostic⁻.moisture.ρq_tot
+
+    u⁻ = ρu⁻ / ρ⁻
+    e⁻ = ρe⁻ / ρ⁻
+    ts⁻ = recover_thermo_state(balance_law, state_prognostic⁻, state_auxiliary⁻)
+    h⁻ = total_specific_enthalpy(ts⁻, e⁻)
+    qt⁻ = ρq_tot⁻ / ρ⁻
+    c⁻ = soundspeed_air(ts⁻)
+
+    ρ⁺ = state_prognositc⁺.ρ
+    ρu⁺ = state_prognostic⁺.ρu
+    ρe⁺ = state_prognostic⁺.ρe
+    ρq_tot⁺ = state_prognostic⁺.moisture.ρq_tot
+
+    u⁺ = ρu⁺ / ρ⁺
+    e⁺ = ρe⁺ / ρ⁺
+    ts⁺ = recover_thermo_state(balance_law, state_prognostic⁺, state_auxiliary⁺)
+    h⁺ = total_specific_enthalpy(ts⁺, e⁺)
+    qt⁺ = ρq_tot⁺ / ρ⁺
+    c⁺ = soundspeed_air(ts⁺)
+    ũ = RoeAverage(ρ⁻, ρ⁺, u⁻, u⁺)
+    e_tot = RoeAverage(ρ⁻, ρ⁺, e⁻, e⁺)
+    h̃ = RoeAverage(ρ⁻, ρ⁺, h⁻, h⁺)
+    qt = RoeAverage(ρ⁻, ρ⁺, qt⁻, qt⁺)
+    ρ = sqrt(ρ⁻ * ρ⁺)
+    e_int⁻ = internal_energy(
+        ρ⁻,
+        ρe⁻,
+        ρu⁻,
+        gravitational_potential(balance_law, state_auxiliary⁻),
+    )
+    e_int⁺ = internal_energy(
+        ρ⁺,
+        ρe⁺,
+        ρu⁺,
+        gravitational_potential(balance_law, state_auxiliary⁺),
+    )
+    e_int = RoeAverage(ρ⁻, ρ⁺, e_int⁻, e_int⁺)
+    ts = PhaseEquil(
+        param_set,
+        e_int,
+        ρ,
+        qt,
+        balance_law.moisture.maxiter,
+        balance_law.moisture.tolerance,
+    )
+    c̃ = sqrt(RoeAverage(ρ⁻, ρ⁺, c⁻^2, c⁺^2))
+    _cv_m = cv_m(ts)
+    R_m = gas_constant_air(ts)
+    _cp_m = cp_m(ts)
+    gamma = _cp_m / _cv_m
+    # chosen by fair dice roll
+    # guaranteed to be random
+    ω = FT(π) / 3
+    δ = FT(π) / 5
+    random_unit_vector = SVector(sin(ω) * cos(δ), cos(ω) * cos(δ), sin(δ))
+    # tangent space basis
+    τ1 = random_unit_vector × normal_vector
+    τ2 = τ1 × normal_vector
+    ũᵀn⁻ = u⁻' * normal_vector
+    ũᵀn⁺ = u⁺' * normal_vector
+    ũᵀn = ũ' * normal_vector
+    ũc̃⁻ = ũ + c̃ * normal_vector
+    ũc̃⁺ = ũ - c̃ * normal_vector
+    e_kin_pot = h̃ - _e_int_v0 * qt - _cp_m * c̃^2 / R_m
+    if (numerical_flux.LM == true)
+        Mach⁺ = sqrt(u⁺' * u⁺) / c⁺
+        Mach⁻ = sqrt(u⁻' * u⁻) / c⁻
+        Mach = (Mach⁺ + Mach⁻) / 2 #RoeAverage(ρ⁻, ρ⁺, Mach⁻, Mach⁺)
+        c̃_LM = c̃ * min(Mach * sqrt(4 + (1 - Mach^2)^2) / (1 + Mach^2), 1)
+    else
+        c̃_LM = c̃
+    end
+    #Standard Roe
+    Λ = SDiagonal(
+        abs(ũᵀn - c̃_LM),
+        abs(ũᵀn),
+        abs(ũᵀn),
+        abs(ũᵀn),
+        abs(ũᵀn + c̃_LM),
+        abs(ũᵀn),
+    )
+    if (numerical_flux.HH == true)
+        Λ = SDiagonal(
+            max(
+                abs(ũᵀn - c̃_LM),
+                max(
+                    0,
+                    ũᵀn - c̃_LM - (u⁻' * normal_vector - c⁻),
+                    u⁺' * normal_vector - c⁺ - (ũᵀn - c̃_LM),
+                ),
+            ),
```

Comment:
Can we precompute these components to battle the formatter?

Reply:
This is written this way to be consistent with the formulation but yes, we could precompute this into separate quantities.